src/openrct2/rct12/RCT12.h

```diff
@@ -276,6 +276,7 @@ struct RCT12TileElement : public RCT12TileElementBase
     uint8_t pad_04[4];
     template<typename TType, RCT12TileElementType TClass> TType* as() const
     {
+        // TODO: CAST-IMPROVEMENT-NEEDED
         return (RCT12TileElementType)GetType() == TClass ? (TType*)this : nullptr;
```

Comment:
static_cast<RCT12TileElementType>(GetType()) == TClass ? reinterpret_cast<TType*>(this) : nullptr;

Reply:
Perhaps GetType should just return `RCT12TileElementType`, so we don't need a cast for that.